test/python/circuit/test_circuit_registers.py

```diff
@@ -74,3 +74,81 @@ def test_cregs_circuit(self):
         self.assertEqual(c_regs[0], cr1)
         self.assertEqual(c_regs[1], cr2)
+
+    def test_basic_slice(self):
+        """simple slice test"""
+        qr = QuantumRegister(5)
+        cr = ClassicalRegister(5)
+        self.assertEqual(len(qr[0:3]), 3)
+        self.assertEqual(len(cr[0:3]), 3)
+
+    def test_apply_gate_to_slice(self):
+        """test applying gate to register slice"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        qc.x(qr[0])
+        qc.y(qr)
+        qc.x(qr[0:2])
+        qc.h(qr[slice(0, 9, 2)])
+        self.log.info(qc.qasm())
+
+    def test_apply_barrier_to_slice(self):
+        """test applying barrier to register slice"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        qc.barrier(qr)
+        qc.barrier(qr[0:2])
+        self.log.info(qc.qasm())
+
+    def test_apply_ccx_to_slice(self):
+        """test applying ccx to register slice"""
+        qcontrol = QuantumRegister(10)
+        qcontrol2 = QuantumRegister(10)
+        qtarget = QuantumRegister(5)
+        qtarget2 = QuantumRegister(10)
+        qc = QuantumCircuit(qcontrol, qtarget, qcontrol2, qtarget2)
+        qc.ccx(qcontrol[1::2], qcontrol[0::2], qtarget)
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol[2:0:-1], qcontrol[4:6], qtarget[0:2])
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol, qcontrol2, qtarget2)
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol[0:2], qcontrol[2:4], qcontrol[5:7])
+        self.log.info(qc.qasm())
+
+    def test_cswap_on_slice(self):
+        """test applying cswap to register slice"""
+        qr1 = QuantumRegister(10)
+        qr2 = QuantumRegister(5)
+        qc = QuantumCircuit(qr1, qr2)
+        qc.cswap(qr2[3::-1], qr1[1:9:2], qr1[2:9:2])
+        qc.cswap(qr2[0], qr1[1], qr1[2])
+        qc.cswap([qr2[0]], [qr1[1]], [qr1[2]])
+        self.assertRaises(qiskit.exceptions.QiskitError, qc.cswap, qr2[4::-1],
+                          qr1[1:9:2], qr1[2:9:2])
+
+    def test_apply_ccx_to_non_register(self):
+        """test applying ccx to non-register raises"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        self.assertRaises(qiskit.exceptions.QiskitError, qc.ccx, qc[0:2], qc[2:4], qc[5:7])
+
+    def test_apply_cx_to_non_register(self):
+        """test applying ccx to non-register raises"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        self.assertRaises(qiskit.exceptions.QiskitError, qc.cx, qc[0:2], qc[2:4])
+
+    def test_apply_ch_to_slice(self):
+        """test applying ch to slice"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        qc.ch(qr[0:2], qr[2:4])
+        qc.barrier(qr)
+        qc.ch(qr[0], qr[1])
+        self.log.info(qc.qasm())
```

Comment:
can you assert something in this test? maybe comparing `qc` against another circuit where you apply `ch` gates one by one, rather than use slice.